packages/react-button/src/components/Button/useButtonState.ts

```diff
@@ -47,6 +63,9 @@ export const useButtonState = (draftState: ButtonState) => {
     }
   };
 
-  draftState['aria-disabled'] = disabled /* || disabledFocusable*/;
-  draftState.disabled = draftState.as === 'button' ? disabled /* && !disabledFocusable */ : undefined;
+  // Set the aria-disabled and disabled props correctly.
+  state['aria-disabled'] = disabled /*|| disabledFocusable*/;
+  state.disabled = as === 'button' ? disabled /* && !disabledFocusable*/ : undefined;
```

Comment:
Will `as` ever be undefined, or does it get defaulted to `'button'` somewhere?

Reply:
It gets defaulted in the `useButton` hook before calling `useButtonState`.